src/http/matcher.cc

```diff
@@ -65,7 +66,12 @@ size_t param_matcher::match(const sstring& url, size_t ind, parameters& param) {
         }
         return sstring::npos;
     }
-    param.set(_name, url.substr(ind, last - ind));
+    sstring path;
+    if (http::internal::url_decode(url.substr(ind, last - ind), path)) {
```

Comment:
@nyh , `url_decode` doesn't do path decoding correctly, see #2125 (https://github.com/scylladb/seastar/pull/2125/commits/8b7f8a3badbeeec0096d24ab637c87311c9d4e16)

Reply:
Yes, I noticed the incorrect "+" decoding added by @bhalevy in 799a9aacb53db4f13c35caecf5071cf078b37159. I somehow accepted this when I reviewed it (https://github.com/scylladb/seastar/pull/1531) - saying that I "am not happy" didn't prevent me from merging it :-(
I thought that this was a pre-existing problem that my patch didn't need to solve, but perhaps you're right it should - especially when it comes to the path when that HTML-only "+" decoding should **never** be done.